### DISCUSSION TITLE
How to delete object if erased by Eraser?

### DISCUSSION THREAD

**Discussion by Saqib92** [fabricjs/fabric.js#8586] [Q&A] [answered]
How can i delete an object / image if erased by Eraser? 
is there any way to get completely Erased object. current if i erase object it is still available on canvas. 
I have Free drawing object, Images, Shapes etc.

**Comment by ShaMan123** [comment, marked as answer]
duplicate,
there is a thread about this somewhere

**Reply by Saqib92** [reply]
can you please give link to that thread? I tried searching it but not able to find it.

**Reply by ShaMan123** [reply]
I can't find it as well
Long story short, I don't see an easy solution in terms of performance.
Checking image data is a possibility.
Maybe scaling down first might help with some aliasing technique, maybe image smoothing.
For sure you could mask the object so if there is anything left it renders black and then check that.
If you find out anything do share

**Reply by LiamPerson** [reply]
Do you have that thread link?

**Reply by Saqib92** [reply]
No

**Reply by luongtu996** [reply]
https://github.com/fabricjs/fabric.js/pull/6994#issuecomment-829075360

**Reply by ShaMan123** [reply]
I am not experienced with filters/WebGL but I wonder if `BlendColor` or something like that can be used to understand if the object is transparent with good perf results
